packages/node-opentelemetry/src/instrumentations/http.ts

```diff
@@ -80,240 +80,269 @@ export const _handleHttpOutgoingClientRequest = (
-      } else {
-        chunks.push(chunk);
-      }
+      const headers =
+        splitCommaSeparatedStrings(httpCaptureHeadersClientResponse) ??
+        response.headers;
+      headerCapture('response', Object.keys(headers))(
+        span,
+        (header) => headers[header],
+      );
     } catch (e) {
-      hdx(`error in response.on('data'): ${e}`);
+      hdx(`error parsing outgoing-response headers in responseHook: ${e}`);
     }
-  }).on('end', () => {
-    try {
-      if (chunks.length > 0) {
-        const buffers = Buffer.concat(chunks);
-        let body = buffers.toString('utf8');
-        const isGzip = response.headers['content-encoding'] === 'gzip';
-        if (isGzip) {
-          body = zlib.gunzipSync(buffers).toString('utf8');
-        }
-        if (shouldRecordBody(body)) {
-          span.setAttribute('http.response.body', body);
+  }
+
+  /* Capture Body */
+  if (networkBodyCapture) {
+    const chunks = [];
+    const pt = new PassThrough();
+    pt.on('data', (chunk) => {
+      try {
+        if (typeof chunk === 'string') {
+          chunks.push(Buffer.from(chunk));
         } else {
-          span.setAttribute('http.response.body', SENSITIVE_DATA_SUBSTITUTE);
+          chunks.push(chunk);
         }
+      } catch (e) {
+        hdx(`error in response.on('data'): ${e}`);
       }
-    } catch (e) {
-      hdx(`error in response.on('end'): ${e}`);
-    }
-  });
-  interceptReadableStream(response, pt);
+    }).on('end', () => {
+      try {
+        if (chunks.length > 0) {
+          const buffers = Buffer.concat(chunks);
+          let body = buffers.toString('utf8');
+          const isGzip = response.headers['content-encoding'] === 'gzip';
+          if (isGzip) {
+            body = zlib.gunzipSync(buffers).toString('utf8');
+          }
+          if (shouldRecordBody(body)) {
+            span.setAttribute('http.response.body', body);
+          } else {
+            span.setAttribute('http.response.body', SENSITIVE_DATA_SUBSTITUTE);
+          }
+        }
+      } catch (e) {
+        hdx(`error in response.on('end'): ${e}`);
+      }
+    });
+    interceptReadableStream(response, pt);
+  }
 };
 
 export const getHyperDXHTTPInstrumentationConfig = ({
+  networkHeadersCapture,
+  networkBodyCapture,
   httpCaptureBodyKeywordsFilter,
   httpCaptureHeadersClientRequest,
   httpCaptureHeadersClientResponse,
   httpCaptureHeadersServerRequest,
   httpCaptureHeadersServerResponse,
 }: {
+  networkHeadersCapture: boolean;
+  networkBodyCapture: boolean;
   httpCaptureBodyKeywordsFilter?: string;
   httpCaptureHeadersClientRequest?: string;
   httpCaptureHeadersClientResponse?: string;
   httpCaptureHeadersServerRequest?: string;
   httpCaptureHeadersServerResponse?: string;
 }) => {
-  const shouldRecordBody = getShouldRecordBody(httpCaptureBodyKeywordsFilter);
+  const shouldRecordBody =
+    networkBodyCapture && getShouldRecordBody(httpCaptureBodyKeywordsFilter);
```

Comment:
the type here is incorrect. this should be something like
```
const shouldRecordBody =  networkBodyCapture ? getShouldRecordBody(httpCaptureBodyKeywordsFilter) : (body: string) => false
```
Also since we have passed `shouldRecordBody`, we don't need to pass `networkBodyCapture` to the downstream, right ?

Reply:
Thanks for the review, changes made